exec/java-exec/src/main/java/org/apache/drill/exec/store/bson/BsonRecordReader.java

```diff
@@ -178,6 +178,11 @@ private void writeToListOrMap(BsonReader reader, final MapOrListWriterImpl write
         writeTimeStamp(time, writer, fieldName, isList);
         atLeastOneWrite = true;
         break;
+      case DECIMAL128:
+        double readDoubleAsDecimal128 = reader.readDecimal128().doubleValue();
+        writeDouble(readDoubleAsDecimal128, writer, fieldName, isList);
```

Comment:
Drill has VarDecimal data type, please use it instead of double.

Reply:
Resovle it